winch/codegen/src/isa/x64/masm.rs

```diff
@@ -337,14 +339,52 @@ impl Masm for MacroAssembler {
         self.asm.cmp(src.into(), dst, size);
         self.asm.setcc(kind, dst);
     }
+
+    fn clz(&mut self, src: Reg, dst: Reg, size: OperandSize) {
+        if self.flags.has_lzcnt() {
+            self.asm.lzcnt(src, dst, size);
+        } else {
+            let scratch = regs::scratch();
+
+            // Use the following approach:
+            // dst = size.num_bits() - bsr(src) - is_not_zero
+            //     = -(bsr(src) + is_not_zero) + size.num_bits().
```

Comment:
I'm not sure how you feel about using comments to write out the algorithm in a higher level syntax. I think plain English is too abstract. And I found having a note of what the formula looks like with multiple operations on the same line, and then an equivalent formula that can be translated to x86 asm to be easier to reason about than trying to write the asm without something higher level to use as a reference.